File: app/src/main/java/com/android_cademy/starwarstinder/ResourcesUtil.java
package com.android_cademy.starwarstinder;

import android.content.Context;
import android.content.res.AssetManager;
import android.util.Log;
import com.android_cademy.starwarstinder.model.Profile;
import com.google.gson.Gson;
import com.google.gson.GsonBuilder;
import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStream;
import java.io.InputStreamReader;
import java.io.Reader;
import java.io.StringWriter;
import java.io.UnsupportedEncodingException;
import java.io.Writer;
import java.util.ArrayList;
import java.util.List;
import org.json.JSONArray;

public class ResourcesUtil {

  public static List<Profile> loadProfiles(Context context) {
    try {
      GsonBuilder builder = new GsonBuilder();
      Gson gson = builder.create();
      JSONArray array = new JSONArray(loadJSONFromAsset(context));
      List<Profile> profileList = new ArrayList<>();
      for (int i = 0; i < array.length(); i++) {
        Profile profile = gson.fromJson(array.getString(i), Profile.class);
        profileList.add(profile);
      }
      return profileList;
    } catch (Exception e) {
      e.printStackTrace();
      return null;
    }
  }

  private static String loadJSONFromAsset(Context context) {
    String jsonString = null;
    Writer writer = new StringWriter();
    char[] buffer = new char[1024];
    try {
      try (InputStream is = context.getResources().openRawResource(R.raw.profiles)) {
        Reader reader = new BufferedReader(new InputStreamReader(is, "UTF-8"));
        int n;
        while ((n = reader.read(buffer)) != -1) {
          writer.write(buffer, 0, n);
        }
      } catch (UnsupportedEncodingException e) {
        e.printStackTrace();
      } finally {
        jsonString = writer.toString();
      }
    } catch (IOException e) {
      e.printStackTrace();
    }
    return jsonString;
  }
}

File: app/src/main/java/com/android_cademy/starwarstinder/ResourcesUtil.kt
package com.android_cademy.starwarstinder

import android.content.Context
import com.android_cademy.starwarstinder.model.Profile
import com.google.gson.GsonBuilder
import org.json.JSONArray
import java.io.BufferedReader
import java.io.InputStreamReader
import java.io.StringWriter
import java.util.ArrayList

class ResourcesUtil {

    companion object {
        fun loadProfiles(context: Context): List<Profile>? {
            val builder = GsonBuilder()
            val gson = builder.create()
            val array = JSONArray(loadJSONFromAsset(context))
            val profileList = ArrayList<Profile>()
            for (i in 0 until array.length()) {
                val profile = gson.fromJson(array.getString(i), Profile::class.java)
                profileList.add(profile)
            }
            return profileList

        }

        private fun loadJSONFromAsset(context: Context): String {
            val writer = StringWriter()
            val buffer = CharArray(1024)
            context.resources.openRawResource(R.raw.profiles).use { `is` ->
                val reader = BufferedReader(InputStreamReader(`is`, "UTF-8"))
                var n = reader.read(buffer)
                while (n != -1) {
                    writer.write(buffer, 0, n)
                    n = reader.read(buffer)
                }
            }
            return writer.toString()
        }
    }
}
